src/lib/peer.js

```diff
@@ -79,9 +91,11 @@ function Peer(config) {
 
     function router(envelope) {
         var nextHop = self.fingerTable.bestCandidate(envelope.dstId);
-        log('nextHop:', nextHop, envelope);
-        if (nextHop === self.peerId.toHex()) {
+        console.log('nextHop:', nextHop, envelope);
+        if (nextHop === self.peerId.toHex() && envelope.dstId === self.peerId.toHex()) {
```

Comment:
Now you are confusing me ;-)
The Chord approach, at least as I understand it, performs the following steps for an incoming message, please correct me if I'm wrong:
 1.  Can I handle the message myself? (Is the destination equal to my ID)
 2.  If not, send it to the peer in the finger table that is closest to the destination ID. (So I think this is what you mean by traversing the ring)
 3.  Same steps are performed at the next peer.

So there is no need for a full traversal of the ring to determine if a certain peer is available or not.
Coming back to my former example, there is a certain peer in the network (and it is not neccesserily the origin peer) which should have a direct link to peer 4 (e.g. peer3).
If this peer calls the `fingerTable.bestCandidate(...)` method, again the same peer is returned (as peer4 is not connected), and the message is delivered in the message event as it was a message for this particular peer (peer3).
That is exactly what I want to avoid with this change request